src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Values/CultureValue.cs

```diff
@@ -0,0 +1,53 @@
+using System.Globalization;
+using System.Text.Encodings.Web;
+using Fluid;
+using Fluid.Values;
+using OrchardCore.Localization;
+
+namespace OrchardCore.DisplayManagement.Liquid.Values;
+
+internal sealed class CultureValue : FluidValue
+{
+    private readonly CultureInfo _culture;
+
+    public CultureValue() => _culture = CultureInfo.CurrentUICulture;
```

Comment:
It's the same thing, a single FluidValue is registered, so this code will run only once. It's ok to register a single value, but you need to change the switches to the same code as before the PR:

```c#
nameof(CultureInfo.NativeName) => new StringValue(CultureInfo.CurrentUICulture.NativeName)
```

This way every time the code is invoked it will use the ambient culture of the request

Reply:
Got it

Reply:
I wrote the changes I had in mind, tell me if that works

Reply:
I already did something similar locally but I stuck on getting the culture from within `ToStringValue()`